timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/LeadershipComponents.java

```diff
@@ -54,15 +65,44 @@ public class LeadershipComponents {
         this.healthCheckPingers = healthCheckPingers;
     }
 
-    public <T> T wrapInLeadershipProxy(Client client, Class<T> clazz, Supplier<T> delegateSupplier) {
+    private <T> T wrapInLeadershipProxy(
+            Class<T> clazz, Supplier<T> delegateSupplier, LeadershipCoordinator leadershipCoordinator) {
+        T instance = AwaitingLeadershipProxy.newProxyInstance(clazz, delegateSupplier, leadershipCoordinator);
+        // this is acceptable since the proxy returned implements Closeable and needs to be closed
+        closer.register((Closeable) instance);
+        return instance;
+    }
+
+    public LeadershipServices createServices(
+            Client client,
+            Supplier<AsyncTimelockService> asyncTimelockServiceSupplier,
+            Supplier<LockService> lockServiceSupplier) {
         LeadershipContext context = getOrCreateNewLeadershipContext(client);
-        T instance = AwaitingLeadershipProxy.newProxyInstance(clazz, delegateSupplier, context.leadershipCoordinator());
+        TaggedMetricRegistry metrics = context.leadershipMetrics().taggedMetrics();
+        Function<InvocationContext, Map<String, String>> tagger =
+                context.leadershipMetrics().suspectedLeaderTag();
+        return ImmutableLeadershipServices.of(
+                InstrumentedAsyncTimelockService.builder(wrapInLeadershipProxy(
+                                AsyncTimelockService.class,
+                                asyncTimelockServiceSupplier,
+                                context.leadershipCoordinator()))
+                        .withHandler(AtlasDbMetrics.taggedMetricsHandler(metrics, AsyncTimelockService.class, tagger))
+                        .withPerformanceTraceLogging()
```

Comment:
Are we certain this does not have any performance implications? The wrapper service for these tends to be very prone to overhead

Reply:
I'm happy to remove this as I don't think we have enabled this for time lock in a long time, if ever.

Reply:
removed `.withPerformanceTraceLogging()`

Reply:
created https://github.com/palantir/atlasdb/pull/7164 to remove a couple other places that don't need perf trace logging (N.B. this is different from producing tracing spans).